File: MultiModalSample/Objecitve-C/MultiModalSample/TravelByViewController.m
//
//  TravelByViewController.m
//  MultiModalSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

#import "TravelByViewController.h"

@implementation TravelByViewController

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
    cell.accessoryType = UITableViewCellAccessoryCheckmark;
    
    
    [_delegate didSelectRowAtIndex:indexPath.row];
    
}

- (void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView cellForRowAtIndexPath:indexPath];
    cell.accessoryType = UITableViewCellAccessoryNone;
    
    [_delegate didDeselectRowAtIndex:indexPath.row];
}

@end

File: MultiModalSample/Swift/MultiModalSample/TravelByViewController.swift
//
//  TravelByViewController.swift
//  MultiModalSample
//
//  Copyright © 2559 Globtech. All rights reserved.
//

import UIKit
import NOSTRASDK
fileprivate func < <T : Comparable>(lhs: T?, rhs: T?) -> Bool {
  switch (lhs, rhs) {
  case let (l?, r?):
    return l < r
  case (nil, _?):
    return true
  default:
    return false
  }
}

fileprivate func > <T : Comparable>(lhs: T?, rhs: T?) -> Bool {
  switch (lhs, rhs) {
  case let (l?, r?):
    return l > r
  default:
    return rhs < lhs
  }
}


protocol TravelByDelegate {
    func didFinishSelectTravelMode(_ travels: [NTMultiModalTransportMode]);
}

class TravelByViewController: UITableViewController {

    var delegate: TravelByDelegate?;
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    override func didMove(toParentViewController parent: UIViewController?) {
        if parent == nil {
            var travels:[NTMultiModalTransportMode] = []
            
            if tableView.indexPathsForSelectedRows?.count > 0 {
                for indexPath in tableView.indexPathsForSelectedRows! {
                    travels.append(NTMultiModalTransportMode(rawValue: indexPath.row)!);
                }
                delegate?.didFinishSelectTravelMode(travels);
            }

            
        }
    }
    

    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let cell = tableView.cellForRow(at: indexPath);
        cell?.accessoryType = .checkmark;
    }
    
    override func tableView(_ tableView: UITableView, didDeselectRowAt indexPath: IndexPath) {
        let cell = tableView.cellForRow(at: indexPath);
        cell?.accessoryType = .none;
    }
    
}
